applications/ParticleMechanicsApplication/custom_conditions/particle_based_conditions/mpm_particle_penalty_dirichlet_condition.cpp

```diff
@@ -203,13 +203,41 @@ void MPMParticlePenaltyDirichletCondition::CalculateAll(
         {
             noalias(rLeftHandSideMatrix)  += prod(trans(shape_function), shape_function);
             rLeftHandSideMatrix  *= m_penalty * this->GetIntegrationWeight();
+
         }
 
         if ( CalculateResidualVectorFlag == true )
         {
             noalias(rRightHandSideVector) -= prod(prod(trans(shape_function), shape_function), gap_function);
             rRightHandSideVector *= m_penalty * this->GetIntegrationWeight();
         }
+
+        if (Is(SLIP)){
+            if (CalculateStiffnessMatrixFlag == true){
+                GetRotationTool().Rotate(rLeftHandSideMatrix, rRightHandSideVector, GetGeometry());
+            } else {
+                GetRotationTool().RotateRHS(rRightHandSideVector, GetGeometry());
+            }
+
+            if (CalculateStiffnessMatrixFlag == true) {
+                MatrixType temp_matrix = ZeroMatrix(rLeftHandSideMatrix.size1(),rLeftHandSideMatrix.size2());
```

Comment:
Allocating such a large matrix on the fly can have a performance impact.
Maybe you can reuse an existing variable or modify in-place?

Reply:
I'll modify it in-place instead.